cloud-provider-snapshot-restore-jobs/cmd/resource/resource.go

```diff
@@ -0,0 +1,217 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	matlasClient "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+const (
+	automated = "automated"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	deliveryType := currentModel.DeliveryType.Value()
+	if deliveryType == nil || (*deliveryType != "automated" && *deliveryType != "download") {
```

Comment:
Agree with @thetonymaster that `"automated"` and `"download"` should be both made constants